modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -2122,6 +2122,7 @@ void GDScriptAnalyzer::resolve_for(GDScriptParser::ForNode *p_for) {
 	}
 
 	GDScriptParser::DataType variable_type;
+	GDScriptParser::DataType pair_variable_type;
```

Comment:
```suggestion
	GDScriptParser::DataType first_variable_type;
	GDScriptParser::DataType second_variable_type;
```

`pair` is confusing because it is the type of the second variable, not the type of the pair as a whole. `key` and `value` won't work because arrays iterate over values. The same in other places.